Compilation fix for systems without MAXHOSTNAMELEN (e.g., Hurd); but the
use of MAXHOSTNAMELEN is incorrect here anyway.  (bug #387665,
http://bugs.ntp.org/978)

Index: ntp-4.2.4p0+dfsg/ntpd/ntp_intres.c
===================================================================
--- ntp-4.2.4p0+dfsg.orig/ntpd/ntp_intres.c	2007-04-10 10:48:59.000000000 +0200
+++ ntp-4.2.4p0+dfsg/ntpd/ntp_intres.c	2007-04-10 11:21:40.000000000 +0200
@@ -526,10 +526,10 @@
 			msyslog(LOG_INFO, "findhostaddr: Resolving %s>",
 				stoa(&entry->peer_store));
 #endif
-		entry->ce_name = emalloc(MAXHOSTNAMELEN);
+		entry->ce_name = emalloc(NI_MAXHOST);
 		error = getnameinfo((const struct sockaddr *)&entry->peer_store,
 				   SOCKLEN(&entry->peer_store),
-				   (char *)&entry->ce_name, MAXHOSTNAMELEN,
+				   (char *)&entry->ce_name, NI_MAXHOST,
 				   NULL, 0, 0);
 	}
 #ifdef DEBUG
